packages/shared/components/atoms/AmountBox.svelte

```diff
@@ -1,23 +1,79 @@
-    export let amount: string
+    export let amount: string = ''
     export let fiatAmount: string = ''
+
+    let displayedAmount: string[] = [amount]
+    let tokenAmountElement: HTMLElement = null
+    let isTooltipVisible = false
+
+    $: {
+        if (amount.length > 12) {
+            displayedAmount = breakAmountIntoLines(amount)
+        } else {
+            displayedAmount = [amount]
+        }
+    }
+
+    function showTooltip(): void {
+        const hasDecimal = amount.includes(getDecimalSeparator())
+        const hasMoreThanTwoDecimalNumbers = hasDecimal && amount.split(getDecimalSeparator())[1].length > 2
+        if (amount.length > 12 && hasDecimal && hasMoreThanTwoDecimalNumbers) {
+            isTooltipVisible = true
+        }
+    }
+
+    function hideTooltip(): void {
+        isTooltipVisible = false
+    }
+
+    function breakAmountIntoLines(amount: string): string[] {
+        const NUMBERS_PER_THOUSAND_GROUP = 3
+        const LINES = 2
+
+        const amountWithTwoDecimalNumbers = amount.slice(0, amount.indexOf(getDecimalSeparator()) + 3)
+        const amountLengthWithoutSeparator = amountWithTwoDecimalNumbers.split(',').join('').length
+        const thousandSeparatorsIndexes: number[] = Array.from(amountWithTwoDecimalNumbers).reduce(
+            (acc, char, i) => (char === getGroupSeparator() ? [...acc, i] : acc),
+            []
+        )
+        const thousandGroups = Math.ceil(amountLengthWithoutSeparator / NUMBERS_PER_THOUSAND_GROUP / LINES)
+
+        const result: string[] = []
+        result.push(amountWithTwoDecimalNumbers.substring(0, thousandSeparatorsIndexes[thousandGroups - 1] + 1))
+        result.push(amountWithTwoDecimalNumbers.substring(result[0].length, amountWithTwoDecimalNumbers.length))
+
+        return result
+    }
 </script>
 
 <amount class="flex flex-col items-center">
-    <div class="flex flex-row space-x-3">
+    <token-amount
+        bind:this={tokenAmountElement}
+        class="flex flex-row space-x-3"
+        on:mouseenter={showTooltip}
+        on:mouseleave={hideTooltip}
+    >
         <AssetIcon {asset} />
-        <div class="flex flex-row flex-wrap justify-center items-baseline space-x-0.1">
-            <Text type={TextType.h1} fontWeight={FontWeight.semibold}>{amount}</Text>
-            {#if unit}
-                <Text type={TextType.h4} classes="ml-1" fontWeight={FontWeight.medium}>{unit}</Text>
-            {/if}
+        <div class="flex flex-col flex-wrap justify-center items-baseline space-x-0.1">
+            {#each displayedAmount as line, index}
+                <div class="flex items-baseline">
+                    <Text type={TextType.h1} fontWeight={FontWeight.semibold}>{line}</Text>
+                    {#if unit && index === displayedAmount.length - 1}
+                        <Text type={TextType.h4} classes="ml-1" fontWeight={FontWeight.medium}>{unit}</Text>
+                    {/if}
+                </div>
+            {/each}
         </div>
```

Comment:
cant we rewrite this with some combination of `flex-wrap` and `break-words`?

Reply:
I tried to do it but I wasn't able to do it, I've also previously tried with `white-space` and it didn't work :(
But if you have a solution I would like to learn from it

Reply:
so one solution would be to do it the following way:
```
<Text type={TextType.h1} fontWeight={FontWeight.semibold}>{amount}
   {#if unit }
        <span  class="ml-1">{unit}</span>
   {/if}
</Text>
```

The `span` would need to be styled manually

Reply:
another solution would be to give the `h1` containing the amount a `max-widht` + `break-word`

Reply:
those solutions have different results, but both valid IMO. So either the unit is always on the right (sol.2) or the unit is part of the unit and added after the last digit, like you implemented (sol.1). Feel free to choose

Reply:
Thank you for your suggestions, I've found a new solution based on yours :bulb: 
```svelte
<Text type={TextType.h1} fontWeight={FontWeight.semibold} classes="whitespace-pre items-baseline">
  {displayedAmount.join('\n')}
  {#if unit}
    <Text type={TextType.h4} classes="inline" fontWeight={FontWeight.medium}>{unit}</Text>
  {/if}
</Text>
```